File: responses/RVM3S9-UP8/find_maximum_product.js
// you can write to stdout for debugging purposes, e.g.
// console.log('this is a debug message');

function solution(X, B) {
    // write your code in JavaScript
    counter = 1;
    for (i=1; i<B.length; i++) {
        if (B[i] == (B[i-1] + 1)) {
            counter++;
        } else {
            counter = 1;
        }
    }
    max_prod = X * counter;
    return max_prod
}

File: responses/7A2QDT-RKD/find_maximum_product.py
# you can write to stdout for debugging purposes, e.g.
# print("this is a debug message")
from functools import reduce
def solution(X, B):
    # write your code in Python
    K = len(B)
    if K == 0 or X > K:
        return 0
    else:
        #choose first X number as initialization
        max_prod = reduce((lambda x, y: x * y), B[0:X])
        for i in range(X, len(B)):
            if B[i] > B[i-X]:
                max_prod = max_prod/B[i-X]*B[i]
        return int(max_prod)
                